src/components/molecules/Pagination.tsx

```diff
@@ -40,7 +44,9 @@ export default function Pagination({
       marginPagesDisplayed={smallViewport ? 0 : 1}
       pageRangeDisplayed={smallViewport ? 3 : 6}
       onPageChange={(data) => onPageChange(data.selected)}
-      hrefBuilder={(pageIndex) => hrefBuilder(pageIndex)}
+      hrefBuilder={(pageIndex) => {
```

Comment:
There shouldn't be anything related to url here, this is a simple pagination component, it's should work with server side pagination and client side. `hrefBuilder` should be called in the parent component in `onPageChange`

Reply:
For example `AssetQueryList` is also used in `PublishedList` and now it's not working. If i think more about it `AssetQueryList` should be called just `AssetList` and accept a list of `DDO` and `Pagination` should do just that. The rest of the logic is done in the parent component , for example in search : you do the query, pagination logic and the url stuff in the search component and then pass the query results  (the `DDO` list) to the `AssetList` and what params you need to the pagination

Reply:
the function will be called inside `AssetQueryList`, I just tried to make it optional inside `Pagination`. But I can move that check inside `AssetQueryList`

Reply:
so I called it `AssetQueryList` back then to make it clear this always accepts OUR search query. And that always has pagination info.

We just need to decide what use case this should support, it seems the work here is to ultimately tackle #351. Based on that it seems we should figure out how `AssetQueryList` can always have a pagination inside the component, and not just on the search page

Reply:
We can leave the pagination in there, and the `AssetList` component can have `assets: DDO[]` , `page`, `totalPages`, `onPageChange` . This way we can use it in multiple ways, we can use it on homepage as well, instead of showing two rows each subsection can be paginated and the use can just browse with preset filters, for the algos, in the history, and others. We need to decouple this component from the data source

Reply:
this sounds good for me. Maybe we should add a property like `showPagination` or `hidePagination` in case we don't want the pagination to be displayed

Reply:
yes, and except for `assets` and maybe `showPagination` all the other are optional, maybe you just want to display 4 items and that's it.

Reply:
ok, then i'll also rename the `AssetQueryList` to `AssetList`